File: PHP/Pc.php
<?php
//Mengimport kelas Processor, Ram, dan Disk
include "Processor.php";
include "Ram.php";
include "Disk.php";

class Pc {
    //atribut
    private $processorObj;
    private $ramObj;
    private $diskObj;
    private $totalPrice;

    //constructor
    function __construct($processorObj, $ramObj, $diskObj) {
        $this->processorObj=$processorObj;
        $this->ramObj=$ramObj;
        $this->diskObj=$diskObj;
    }

    //Setter Getter
    function  setProcessorObj($processorObj) { $this->$processorObj=$processorObj; }
    function  setRamObj($ramObj) { $this->ramObj=$ramObj; }
    function  setDiskObj($diskObj) { $this->$diskObj=$diskObj; }
    function  setTotalPrice($pPrice, $rprice, $dPrice)
    {
        $this->totalPrice=$pPrice + $rprice + $dPrice;
    }

    function getProcessorObj() { return $this->processorObj; }
    function getRamObj() { return $this->ramObj; }
    function getDiskObj() { return $this->diskObj; }
    function getTotalPrice() { return $this->totalPrice; }

    //Method
    //Menampilkan atribut Pc
    function  printPc() {
        $this->processorObj->printProcessor();
        $this->ramObj->printRam();
        $this->diskObj->printDisk();
        $this->setTotalPrice($this->processorObj->getPrice(), $this->ramObj->getPrice(), $this->diskObj->getPrice());
        echo "Total Price: " . $this->getTotalPrice() . "<br>";
    }

    //Destructor
    function __destruct() {}
}

File: Java/Pc.java
public class Pc {
    //atribut
    private Processor processorObj;
    private Disk diskObj;
    private Ram ramObj;
    private int totalPrice;

    //constructor
    Pc() {}
    Pc(Processor processorObj, Ram ramObj, Disk diskObj) {
        this.processorObj=processorObj;
        this.ramObj=ramObj;
        this.diskObj=diskObj;
    }

    //Setter Getter
    public void setProcessorObj(Processor processorObj) { this.processorObj=processorObj; }
    public void setRamObj(Ram ramObj) { this.ramObj=ramObj; }
    public void setDiskObj(Disk diskObj) { this.diskObj=diskObj; }
    public void setTotalPrice(int pPrice, int rPrice, int dPrice) {
        this.totalPrice=pPrice + rPrice + dPrice;
    }

    public Processor getProcessorObj() { return this.processorObj; }
    public Ram getRamObj() { return this.ramObj; }
    public Disk getDiskObj() { return this.diskObj; }
    public int getTotalPrice() { return this.totalPrice; }

    //Method
    //Menampilkan atribut PC
    public void printPc() {
        this.processorObj.printProcessor();
        this.ramObj.printRam();
        this.diskObj.printDisk();
        this.setTotalPrice(this.processorObj.getPrice(), this.ramObj.getPrice(), this.diskObj.getPrice());
        System.out.println("Total Price: " + this.getTotalPrice());
    }
}
